src/ice_shelf/MOM_ice_shelf_dynamics.F90

```diff
@@ -249,20 +252,21 @@ subroutine register_ice_shelf_dyn_restarts(G, param_file, CS, restart_CS)
   endif
 
   if (active_shelf_dynamics) then
-    allocate( CS%u_shelf(IsdB:IedB,JsdB:JedB) ) ; CS%u_shelf(:,:) = 0.0
-    allocate( CS%v_shelf(IsdB:IedB,JsdB:JedB) ) ; CS%v_shelf(:,:) = 0.0
+    allocate( CS%u_shelf(Isd:Ied,Jsd:Jed) ) ; CS%u_shelf(:,:) = 0.0
```

Comment:
These are not the right memory declaration calls for variables on the C-grid.  For u_shelf it should be (CS%u_shelf(IsdB:IedB,jsd:jed)) and for v_shelf it should be (CS%v_shelf(isd:ied,JsdB:JedB)) .

Reply:
Bob, thanks for reviewing. I miss-'spoke/typed/etc': u_shelf and v_shelf are moved to h_grid. With such formulation, both symmetric and non-symmetric produce identical results.